File: web/plugins/is_empty.js
import { WebPlugin } from "../engine/plugin.js";

export class IsEmptyPlugin extends WebPlugin {
    constructor() {
        super("文字が空か");
    }

    async execute(line, context) {
        const match = line.match(/^文字が空か\s+「(.*)」$/);
        if (!match) return;

        context.runtime.write(
            match[1].length === 0
        );
    }
}

File: pc/plugins/is_empty.py
import re
from core.plugin import Plugin


class IsEmptyPlugin(Plugin):
    def translate(self, line):
        match = re.fullmatch(r'文字が空か\s+「(.*)」', line)
        if not match:
            return None

        value = match.group(1)
        return f"print({value!r} == '')"